### PR TITLE
SAK-46253 Commons: Added XSS protection to contenteditable

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-46253

### CODE REVIEW

**Review by mpellicer — CHANGES_REQUESTED** [review]
This has to be submitted to Sakai Security instead, we're revealing a fix of a security issue.

If we're using a webjar for this I think it should use the includeWebJar approach at a tool level, we're including the library at a portal level but only one tools uses it for now, the includeWebJar method looks more ideal to me.

The rest looks good to me, thanks for the good work.

File: commons/tool/src/webapp/js/commons.js [hunk 1]
```diff
@@ -59,10 +59,17 @@ commons.switchState = function (state, arg) {
 
             var editor = $('#commons-post-creator-editor');
             var wrapAndInsert = function (link, loadThumbnail, text) {
+
+                    const clean = DOMPurify.sanitize(link, {USE_PROFILES: {html: true}});
+                    if (DOMPurify.removed) {
+                        console.warn("DOMPurify removed some dangerous stuff:");
```
Comment (mpellicer) [inline commons/tool/src/webapp/js/commons.js:65]: Do you think we should use debug instead of warn?

File: library/pom.xml [hunk 2]
```diff
@@ -314,7 +314,12 @@
 			<groupId>org.webjars.npm</groupId>
 			<artifactId>flatpickr</artifactId>
 			<version>4.6.9</version>
-		</dependency>
+        </dependency>
```
Comment (mpellicer) [inline library/pom.xml:317]: Minor, spacing.

**Review by mpellicer — CHANGES_REQUESTED** [review]

**Comment by mpellicer** [comment]
https://github.com/sakaiproject/sakai/security/policy

**Comment by SedueRey** [comment]
@adrianfish Please, could yo submit this PR to Sakai Security instead, please? and close and delete this one.

**Comment by adrianfish** [comment]
@SedueRey Will do, sorry. How do I do that? Where's the repo?

**Pull request closed without merging**